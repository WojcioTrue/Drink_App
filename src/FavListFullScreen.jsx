

const FavListFullScreen = () => {
  return (
    <div>FavListFullScreen</div>
  )
}

export default FavListFullScreen