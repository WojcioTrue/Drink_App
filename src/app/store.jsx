import favouriteListReducer from "../features/favouriteList/favouriteListSlice";
import notificationListReducer from "../features/notificationList/notificationListSlice";
import categoryListReducer from "../features/categoryList/categoryListSlice";
import searchBarReducer from "../features/searchBar/searchBarSlice";
import drinkDataReducer from "../features/drinkInfo/drinkDataSlice";
import ingredientsButtonsReducer from "../features/ingredientSearch/ingredientsButtonsSlice";
import ingredientsDataReducer from "../features/ingredientSearch/ingredientsDataSlice";
import { combineReducers, configureStore } from "@reduxjs/toolkit";


const rootReducer = combineReducers({
  favouriteList: favouriteListReducer,
  notificationList: notificationListReducer,
  categoryList: categoryListReducer,
  searchBar: searchBarReducer,
  drinkData: drinkDataReducer,
  ingredientsButtons: ingredientsButtonsReducer,
  ingredientsData : ingredientsDataReducer
});

export const store = configureStore({
  reducer: rootReducer,
});

export function setupStore(preloadedState) {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
}
