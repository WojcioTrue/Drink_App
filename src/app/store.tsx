import favouriteListReducer from "../features/favouriteList/favouriteListSlice";
import notificationListReducer from "../features/notificationList/notificationListSlice";
import categoryListReducer from "../features/categoryList/categoryListSlice";
import searchBarReducer from "../features/searchBar/searchBarSlice";
import drinkDataReducer from "../features/drinkInfo/drinkDataSlice";
import ingredientsButtonsReducer from "../features/ingredientSearch/ingredientsButtonsSlice";
import ingredientsDataReducer from "../features/ingredientSearch/ingredientsDataSlice";
import { PreloadedState, combineReducers, configureStore } from "@reduxjs/toolkit";

import { StateFromReducersMapObject } from "@reduxjs/toolkit";

export type RootState = StateFromReducersMapObject<typeof rootReducer>


const rootReducer = combineReducers({
  favouriteList: favouriteListReducer,
  notificationList: notificationListReducer,
  categoryList: categoryListReducer,
  searchBar: searchBarReducer,
  drinkData: drinkDataReducer,
  ingredientsButtons: ingredientsButtonsReducer,
  ingredientsData : ingredientsDataReducer
});

export const store = configureStore({
  reducer: rootReducer,
});

export function setupStore(preloadedState?: PreloadedState<RootState>) {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
}

type Store = ReturnType<typeof setupStore>
export type AppDispatch = Store['dispatch']