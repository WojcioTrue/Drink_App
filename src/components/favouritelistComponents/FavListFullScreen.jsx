import { useContext } from "react";
import "../../styles/coctail_list.css";
import { MyContext } from "../../context/ContextComponent";
import CoctailListElement from "../../CoctailListElement";
import Message from "../../Message";
import { AnimatePresence } from "framer-motion";
import { noFavouriteDrinks } from "../../framerStyles/variants";
import { useLocation } from 'react-router-dom';


const FavListFullScreen = () => {
  const { listOfFav } = useContext(MyContext);
  const favouriteList = listOfFav.drinks;
  // check if CoctailListElement is inside favourite_list component,
  // if so return true
  const location = useLocation();
  const inFavourite = location.pathname === "/favourite_list";

  return (
    <div className="list-coctails">
      {favouriteList.length > 0 ? (
        <>
          <h3>FavouriteDrinks:</h3>
          <div className="grid-coctails">
          <AnimatePresence>
            {favouriteList.map((element) => {
              return (
                
                  <CoctailListElement
                    key={element.idDrink}
                    id={element.idDrink}
                    name={element.strDrink}
                    imgSrc={element.strDrinkThumb}
                    useLayout={inFavourite}
                  />
                
              );
            })}
            </AnimatePresence>
          </div>
        </>
      ) : (
        <Message
          animationVariant={noFavouriteDrinks}
          header={"Your list is empty!"}
          text={"Add something to your favourite list."}
          secondText={"Drink responsibly!"}
          img={process.env.PUBLIC_URL + "./img/fav_icon.png"}
        />
      )}
    </div>
  );
};

export default FavListFullScreen;
