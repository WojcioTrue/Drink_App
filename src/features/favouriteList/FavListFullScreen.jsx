import "../../styles/coctail_list.css";
import CoctailListElement from "../../components/coctailListComponents/CoctailListElement";
import Message from "../../sharedComponents/Message";
import { AnimatePresence } from "framer-motion";
import { noFavouriteDrinks } from "../../framerStyles/variants";
import { useLocation } from 'react-router-dom';
import { useSelector } from "react-redux";


const FavListFullScreen = () => {
  const  listOfFavourite  = useSelector(state => state.favouriteList)
  const favouriteList = listOfFavourite;
  // check if CoctailListElement is inside favourite_list component,
  // if so return true
  const location = useLocation();
  const inFavourite = location.pathname === "/favourite_list";
  return (
    <div className="list-coctails">
      {favouriteList.length > 0 ? (
        <>
          <h3>Favourite drinks:</h3>
          <div className="grid-coctails">
          <AnimatePresence>
            {favouriteList.map((element) => {
              return (
                
                  <CoctailListElement
                    key={element.idDrink}
                    id={element.idDrink}
                    name={element.strDrink}
                    imgSrc={element.strDrinkThumb}
                    useLayout={inFavourite}
                  />
                
              );
            })}
            </AnimatePresence>
          </div>
        </>
      ) : (
        <Message
          animationVariant={noFavouriteDrinks}
          header={"Your list is empty!"}
          text={"Add something to your favourite list."}
          secondText={"Drink responsibly!"}
          img={process.env.PUBLIC_URL + "./img/fav_icon.png"}
        />
      )}
    </div>
  );
};

export default FavListFullScreen;
