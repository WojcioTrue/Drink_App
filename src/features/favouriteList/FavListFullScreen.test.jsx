import { cleanup, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import FavListFullScreen from "./FavListFullScreen";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../../app/store";

afterEach(cleanup);

describe("tests for favourite list container full screen", () => {
  test("Basic header content test with empty list", () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <FavListFullScreen />
        </Provider>
      </MemoryRouter>
    );

    const bottleImg = screen.getByRole("img");
    const header = screen.getByText(/Your list is empty!/i);
    const addText = screen.getByText(/Add something to your favourite list./i);
    const responsinilityText = screen.getByText(/Drink responsibly!/i);

    expect(bottleImg.getAttribute("src")).toBe("./img/fav_icon.png");
    expect(header).toBeInTheDocument();
    expect(addText).toBeInTheDocument();
    expect(responsinilityText).toBeInTheDocument();
  });
  // test("snapshot test with empty list", () => {
  //   const { asFragment } = render(
  //     <MemoryRouter>
  //       <Provider store={store}>
  //         <FavListFullScreen />
  //       </Provider>
  //     </MemoryRouter>
  //   );
  //   expect(asFragment(<FavListFullScreen />)).toMatchSnapshot();
  // });
  test("Get favouriteList from store", async () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <FavListFullScreen />
        </Provider>
      </MemoryRouter>
    );
      console.log(store)

    });
  });
