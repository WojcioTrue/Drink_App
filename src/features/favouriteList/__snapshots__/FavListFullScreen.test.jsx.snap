// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for favourite list container full screen snapshot test with empty list 1`] = `
<DocumentFragment>
  <div
    class="list-coctails"
  >
    <div
      class="message__list"
      style="opacity: 0;"
    >
      <span
        class="message__list-img"
      >
        <img
          alt="#"
          src="./img/fav_icon.png"
        />
      </span>
      <section>
        <p
          class="header"
        >
          Your list is empty!
        </p>
        <p>
          Add something to your favourite list.
        </p>
        <p>
          Drink responsibly!
        </p>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
